Mathlib/FieldTheory/SeparableDegree.lean

```diff
@@ -705,3 +705,21 @@ theorem separable_inv (x : E) (hx : (minpoly F x).Separable) : (minpoly F x⁻¹
   separable_of_mem_isSeparable F E <| F⟮x⟯.inv_mem <| mem_adjoin_simple_self F x
 
 end Field
+
+/-- A field is a perfect field (which means that any irreducible polynomial is separable)
+if and only if every separable degree one polynomial splits. -/
+theorem perfectField_iff_splits_of_natSepDegree_eq_one (F : Type*) [Field F] :
+    PerfectField F ↔ ∀ f : F[X], f.natSepDegree = 1 → f.Splits (RingHom.id F) := by
```

Comment:
We have lemmas about `f.natSepDegree = 1` assuming `Irreducible f` ([Irreducible.natSepDegree_eq_one_iff_of_monic'](https://leanprover-community.github.io/mathlib4_docs/Mathlib/FieldTheory/SeparableDegree.html#Irreducible.natSepDegree_eq_one_iff_of_monic') etc.), but without that assumption we should still be able to show `f` has the form `(X ^ p ^ n - C a) ^ m` (and you can require that `a : F` has no `p`th root in F).

Reply:
I should say in the proof of this result I cheated a bit and uses `PerfectRing.ofSurjective`. The criterion of a monic polynomial being separable degree one is not used.

Reply:
I have added the result you mentioned. The proof is very ugly though; improvements are welcome. Added a lot of intermediate results to various files. There is still an intermediate result `test1` which has no good descriptive names.